File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        if (nums.length === 0) return 0;

        let set = new Set(nums);
        let arr = Array.from(set).sort((a, b) => a - b);

        let curr = 1;
        let longest = 1;

        for (let i = 1; i < arr.length; i++) {
            if (arr[i] === arr[i - 1] + 1) {
                curr++; // extend streak
            } else {
                curr = 1; // reset streak
            }

            longest = Math.max(longest, curr); // track max
        }

        return longest;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-4.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        s = set()
        nums.sort()
        for i in range(len(nums)):
            l, r = i + 1, len(nums) - 1
            while l < r:
                total = nums[i] + nums[l] + nums[r]
                if total > 0:
                    r -= 1
                elif total < 0:
                    l += 1
                else:
                    s.add((nums[i], nums[l], nums[r]))
                    r -= 1
                    l += 1
        return [list (i) for i in s]